Improbable/Improbable.CSharpCodeGen/Types.cs

```diff
@@ -158,184 +160,143 @@ public static class Types
 
         public static string GetOptionType(TypeReference type)
         {
-            switch (type.ValueTypeSelector)
+            return type.ValueTypeSelector switch
             {
-                case ValueType.Enum:
-                    return "?";
-                case ValueType.Primitive:
-                    switch (type.Primitive)
-                    {
-                        case PrimitiveType.String:
-                        case PrimitiveType.Bytes:
-                            return string.Empty;
-                        default:
-                            return "?";
-                    }
-
-                case ValueType.Type:
-                    return "?";
-                default:
-                    throw new ArgumentOutOfRangeException();
-            }
+                ValueType.Enum => "?",
+                ValueType.Primitive => (type.Primitive switch
+                {
+                    PrimitiveType.String => string.Empty,
+                    PrimitiveType.Bytes => string.Empty,
```

Comment:
Hm can we not do matching like `PrimitiveType.String || PrimitiveType.Bytes => string.Empty`? Or is that not a C# thing? I can't recall

Reply:
I just did a huge pass to "flatten" all of these nested switches
The C# way is `<case> when <condition>`